src/main/java/io/stargate/sgv2/jsonapi/api/model/command/clause/sort/SortClause.java

```diff
@@ -42,18 +42,28 @@ public boolean hasVectorizeSearchClause() {
             .equals(DocumentConstants.Fields.VECTOR_EMBEDDING_TEXT_FIELD);
   }
 
-  public void validate(CommandContext commandContext) {
-    IndexingProjector indexingProjector = commandContext.indexingProjector();
-    // If nothing specified, everything indexed
-    if (indexingProjector.isIdentityProjection()) {
-      return;
-    }
-    // validate each path in sortExpressions
-    for (SortExpression sortExpression : sortExpressions) {
-      if (!indexingProjector.isPathIncluded(sortExpression.path())) {
-        throw ErrorCode.UNINDEXED_SORT_PATH.toApiException(
-            "sort path '%s' is not indexed", sortExpression.path());
-      }
+  public void validate(CommandContext<?> commandContext) {
+
+    switch (commandContext.schemaObject().type) {
+      case COLLECTION:
+        IndexingProjector indexingProjector =
+            commandContext.asCollectionContext().schemaObject().indexingProjector();
+        // If nothing specified, everything indexed
+        if (indexingProjector.isIdentityProjection()) {
+          return;
+        }
+        // validate each path in sortExpressions
+        for (SortExpression sortExpression : sortExpressions) {
+          if (!indexingProjector.isPathIncluded(sortExpression.path())) {
+            throw ErrorCode.UNINDEXED_SORT_PATH.toApiException(
+                "sort path '%s' is not indexed", sortExpression.path());
+          }
+        }
+        break;
+      default:
+        throw new UnsupportedOperationException(
```

Comment:
As per some earlier notes, we probably want to instead throw `ErrorCode.SERVER_INTERNAL_ERROR.toApiException(....)` for things like this (no more code and easier to into specific source when troubleshooting)

Reply:
Filed #1287 to address this after merging (not yet).